### PR TITLE
ChipButton - should emit when a button is added or removed

### PR DESCRIPTION
**Pull request by cokecc**
solve error #336

### CODE REVIEW

File: src/app/systelab-components/chip-button/chip-button.component.ts [hunk 1]
```diff
@@ -81,6 +90,9 @@ export class ChipButtonComponent {
 			this.buttonList.splice(index, 1);
 			this.buttonList = this.buttonList;
 		}
+		this.buttonRemoved.emit(item);
+		const last = this.buttonList[this.buttonList.length - 1];
+		this.selectItem(last);
```
Comment (olgapuig) [inline src/app/systelab-components/chip-button/chip-button.component.ts]: The last item must be selected only when the removed item was selected. Consider that the user can delete a chip button while another one is selected.

File: src/app/systelab-components/chip-button/chip-button.component.ts [hunk 2]
```diff
@@ -81,6 +90,9 @@ export class ChipButtonComponent {
 			this.buttonList.splice(index, 1);
 			this.buttonList = this.buttonList;
```
Comment (olgapuig) [inline src/app/systelab-components/chip-button/chip-button.component.ts]: Remove this line this.buttonList = this.buttonList;

File: src/app/systelab-components/chip-button/chip-button.component.ts [hunk 3]
```diff
@@ -64,9 +66,16 @@ export class ChipButtonComponent {
...
-		this.buttonList.push({name: this.i18nService.instant('COMMON_NEW'), id: this.buttonList.length + 1, isChecked: false});
-		this.selectItem(this.buttonList[this.buttonList.length - 1]);
-
+		let maxID = 0;
+		if (this.buttonList.length > 0) {
+			maxID = Math.max.apply(Math, this.buttonList.map(function(o) {
+				return o.id;
+			}));
+		}
+		this.buttonList.push({name: this.i18nService.instant('COMMON_NEW'), id: maxID + 1, isChecked: false});
```
Comment (olgapuig) [inline src/app/systelab-components/chip-button/chip-button.component.ts]: I would refactor these 2 lines
```
const item = {name: this.i18nService.instant('COMMON_NEW'), id: maxID + 1, isChecked: false};
this.buttonList.push(item);
```

File: src/app/systelab-components/chip-button/chip-button.component.ts [hunk 4]
```diff
@@ -64,9 +66,16 @@ export class ChipButtonComponent {
 	}
 
 	public addButtonITem() {
-		this.buttonList.push({name: this.i18nService.instant('COMMON_NEW'), id: this.buttonList.length + 1, isChecked: false});
-		this.selectItem(this.buttonList[this.buttonList.length - 1]);
-
+		let maxID = 0;
```
Comment (olgapuig) [inline src/app/systelab-components/chip-button/chip-button.component.ts]: I would refactor the following lines (from 69 to 73) to just the line below

`const maxID = Math.max(...this.buttonList.map(o => o.id), 0);`

Comment (cokecc) [reply src/app/systelab-components/chip-button/chip-button.component.ts]: clever

File: src/app/systelab-components/chip-button/chip-button.component.ts [hunk 5]
```diff
@@ -15,6 +15,8 @@ export class ChipButtonComponent {
 
 	@Output() public changeButton = new EventEmitter();
 	@Output() public selectButton = new EventEmitter();
+	@Output() public buttonRemoved = new EventEmitter();
```
Comment (olgapuig) [inline src/app/systelab-components/chip-button/chip-button.component.ts]: Document the EventEmitters in the .md file

Comment (cokecc) [reply src/app/systelab-components/chip-button/chip-button.component.ts]: added

**Pull request merged**